### DISCUSSION TITLE
Failing 'make -j 8' on MAC M1 machine

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#17942] [Q&A Getting Started] [answered]
I'm trying to install MOOSE on my Macbook Air M1 machine, but am unable to pass the testing. I made sure my home directory was wiped of any hidden bash_session, git, conda, and related folders, and did a completely new installation of MOOSE. I made sure to perform the following:

- Updating conda before installation of MOOSE
- Updating MOOSE once MOOSE was cloned with git
- Performing **'conda init bash'** and successfully activating MOOSE
- Installing **moose-pyhit**
- Peforming **'git clean -xfd'**
- Restarting my shell

Nonetheless, both with the **'make -j 4'** and **'make -j 8'** commands, the system reports numerous errors, the majority coming from the file **'~/miniconda3/envs/moose/bin/../include/c++/v1'** with errors such as **'no member named 'strtod' in the global namespace'**, which included other members such as **atol, size_t, div_t,** and more. Later in the diagnostic message, this error appeared: **'clang frontend command failed due to signal (use -v to see invocation)'**

I hope I have illustrated the issue fully enough. If someone needs additional information, I will happily provide it. I've tried to install MOOSE the past two weeks with various methods, and these sorts of errors have shown up every time.

Thanks!

**Comment by milljm** [comment, marked as answer]
Unfortunately, we don't have any packages built for Apple Silicon as of yet.

The following... is a _very_ early stage of building these packages. The current changes can be found on my `m1-bigsur` branch: https://github.com/milljm/moose/commit/150cd561cdb056129e176d1a9201bb7519b16cb3

You will need to first build/install the moose-mpich package using `conda build` based on the changes in my branch. Then, install PETSc 3.15.0 manually using the same configure options you see in the m1-bigsur branch (I believe I am unable to build superlu, and strumpack).

As for libMesh, it is necessary to 'bootstrap' it and several contribs before running configure:
```bash
export PETSC_DIR=/where/ever/you/installed/it

cd moose/scripts
./update_and_rebuild_libmesh.sh     <----- this will fail. But it grabs the appropriate hash of libMesh
cd ../libmesh
./bootstrap
cd contrib/metaphysicl
./bootstrap
cd ../timpi
./bootstrap
cd ../netcdf/netcdf-c-4.6.2/
autoreconf
cd ../../../build
../configure --with-methods="opt" \
                                   --prefix=/some/location \
                                   --enable-silent-rules \
                                   --enable-unique-id \
                                   --disable-warnings \
                                   --enable-glibcxx-debugging \
                                   --disable-maintainer-mode \
                                   --enable-petsc-hypre-required \
                                   --enable-metaphysicl-required
make
make install
export LIBMESH_DIR=/some/location
```

After all of this, you'll be able to _build_ MOOSE. But it still doesn't run... M1 is receiving more traction, so I suspect we will see some packages being made available sooner rather than later.

**Reply by cticenhour** [reply]
One possible edit to your code snippet. I would replace the lines after PETSc export with 
```
cd moose
git submodule update --init libmesh
cd libmesh
```
Since that's all the libmesh script is doing anyway, and it removes seeing scary failures from the equation.
